Make hashie dependency pessimistic

sj26 opened the pull request:
Allows interoperability with chargify2 gem. :-)

sj26 committed:
Make hashie dependency pessimistic

sj26 committed:
Add bundler, split out version file, pessimistic HTTParty

sj26 committed:
Accept a product_id OR some attributes when migrating subscriptions

sj26 committed:
Use MultiJson

sj26 committed:
Missed out `format :json` to parse as json

sj26 committed:
Actually set false on failure.

cldwalker commented:
Modifying hashie makes sense. Not interested in bundler or multi_json. For future ref, multiple issues should be split up into separate pull requests

sj26 commented:
Sorry, this pull request was only meant to be the first commit but the rest got pulled in as we updated our branch.

I encountered issues adding tests because the dependency on Crack wasn't stated. Seeing as HTTParty already relies on multi_json which uses some faster json implementations if they're available this seemed like a natural move. Plus, HTTParty does it for you now. :-)

We split out the version because loading the gem to grab the version was failing for us in certain scenarios. It seems to be [good practise](http://timelessrepo.com/making-ruby-gems) to have a separate file these days, too.

The bundler stuff was just convenience.

This branch will be updated more eventually when I get to look at it again.

cldwalker commented:
Closing since nothing was ever updated and hashie version has been bumped
